scripts/generate-docs/main.go

```
@@ -26,6 +57,33 @@
+		mapstructure := field.Tag.Get("mapstructure")
+		defaultValue := field.Tag.Get("default")
+		values := strings.Split(field.Tag.Get("values"), ", ")
+		if len(values) == 1 && values[0] == "" {
+			values = []string{}
+		}
+		
+		docFile, err := os.Create(fmt.Sprintf("%s/%s.md", ServerConfigDirectory, field.Name))
+		if err != nil {
+			fmt.Printf("ERR: error creating file: %v\n", err)
+		}
+		defer docFile.Close()
```

golangci-lint (1.64.8)

deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop

(gocritic)